.github/workflows/nightly-builds.yaml

```diff
@@ -8,6 +8,35 @@ on:
 permissions: {}
 
 jobs:
+  test-compile:
+    name: Compile
+    runs-on: ubuntu-20.04
```

Comment:
quite old, maybe we should got through and update the images used in all workflows